tests/test_service.rb

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2016-2017 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/service'
+
+include Cisco
+
+# TestService - Minitest for feature Service node Utility class
+class TestService < CiscoTestCase
+  @skip_unless_supported = 'service'
+
+  def preconfig_service_info
+    path = File.expand_path('../service_info.yaml', __FILE__)
+    skip('Cannot find tests/service_info.yaml') unless File.file?(path)
+    info = YAML.load(File.read(path))
+    valid_info?(info)
+    info
+  end
+
+  def valid_info?(info)
+    skip('tests/service_info.yaml file is empty') unless info
+    msg = 'Missing key in tests/service_info.yaml'
+    %w(install_image install_media).each do |key|
+      skip("#{msg}: #{key}") if info[key].nil?
+    end
+  end
+
+  ###################
+  # Service tests   #
+  ###################
+
+  def test_clear_status
+    Service.clear_status
+    refute(Service.upgraded?)
+  end
+
+  def test_delete
+    shell_command('touch /bootflash/foobar')
+    Service.delete('foobar')
+    assert_raises(CliError) do
+      Service.delete('foobar')
+    end
+  end
+
+  def test_delete_negative
+    assert_raises(CliError) do
+      # Delete a file that doesn't exist
+      Service.delete('foobar')
+    end
+    assert_raises(CliError) do
+      # Delete a file that doesn't exist
+      Service.delete('foobar', 'logflash:')
+    end
+  end
+
+  # def test_delete_boot
+  #   Service.delete_boot
+  # end
+
+  def test_image_version
+    version = Service.image_version
+    assert_match(/^\d.\d\(\d\)\S+\(\d+\)$/, version)
+  end
+
+  def test_upgrade
```

Comment:
@rahushen Just realized that this test will fail if you run it and try to upgrade with the image that is currently booted.  The install all command will still go through a series of checks but then indicate that the install was successful.

Here is the sample output.

```
n9k-109# sh install all status 
There is an on-going installation...
Enter Ctrl-C to go back to the prompt.


Verifying image bootflash:/nxos.7.0.3.I5.0.231.CSCvb69953.bin for boot variable "nxos".
 -- SUCCESS

Verifying image type.
 -- SUCCESS

Preparing "bios" version info using image bootflash:/nxos.7.0.3.I5.0.231.CSCvb69953.bin.
 -- SUCCESS

Preparing "nxos" version info using image bootflash:/nxos.7.0.3.I5.0.231.CSCvb69953.bin.
 -- SUCCESS

Performing module support checks.
 -- SUCCESS

Notifying services about system upgrade.
 -- SUCCESS



Compatibility check is done:
Module  bootable          Impact  Install-type  Reason
------  --------  --------------  ------------  ------
     1       yes  non-disruptive          none  
    22       yes  non-disruptive          none  
    24       yes  non-disruptive          none  
    26       yes  non-disruptive          none  
    28       yes  non-disruptive          none  
    29       yes  non-disruptive          none  
    30       yes  non-disruptive          none  



Images will be upgraded according to following table:
Module       Image                  Running-Version(pri:alt)           New-Version  Upg-Required
------  ----------  ----------------------------------------  --------------------  ------------
     1       lcn9k                               7.0(3)I5(1)           7.0(3)I5(1)            no
     1        bios                       v01.48(00:v01.40(00             v01.48(00            no
    22       lcn9k                               7.0(3)I5(1)           7.0(3)I5(1)            no
    22        bios                       v01.48(00:v01.42(00             v01.48(00            no
    24       lcn9k                               7.0(3)I5(1)           7.0(3)I5(1)            no
    24        bios                       v01.48(00:v01.42(00             v01.48(00            no
    26       lcn9k                               7.0(3)I5(1)           7.0(3)I5(1)            no
    26        bios                       v01.48(00:v01.42(00             v01.48(00            no
    28        nxos                               7.0(3)I5(1)           7.0(3)I5(1)            no
    28        bios     v08.32(10/18/2016):v08.06(09/10/2014)    v08.31(08/26/2016)            no
    29       lcn9k                               7.0(3)I5(1)           7.0(3)I5(1)            no
    29        bios                       v01.48(00:v01.42(00             v01.48(00            no
    30       lcn9k                               7.0(3)I5(1)           7.0(3)I5(1)            no
    30        bios                       v01.48(00:v01.42(00             v01.48(00            no



Install is in progress, please wait.

Performing runtime checks.
 -- SUCCESS

Setting boot variables.
 -- SUCCESS

Performing configuration copy.
 -- SUCCESS

Module 1: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Module 22: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Module 24: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Module 26: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Module 28: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Module 29: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Module 30: Refreshing compact flash and upgrading bios/loader/bootrom.
Warning: please do not remove or power off the module at this time.
 -- SUCCESS

Install has been successful.
n9k-109# 
```

We probably need to update the yaml file to look for both of the following.
`get_value: '/Finishing the upgrade, switch will reboot|Install has been successful/'`